pkg/kubelet/apis/device-plugin/v1alpha1/api.proto

```diff
@@ -0,0 +1,90 @@
+// To regenerate api.pb.go run hack/update-generated-runtime.sh
+syntax = 'proto3';
+
+package deviceplugin;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+option (gogoproto.goproto_stringer_all) = false;
+option (gogoproto.stringer_all) =  true;
+option (gogoproto.goproto_getters_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_unrecognized_all) = false;
+
+
+service Registration {
+	rpc Register(RegisterRequest) returns (RegisterResponse) {}
+}
+
+service DevicePlugin {
+	rpc ListWatch(Empty) returns (stream Device) {}
+	rpc Allocate(AllocateRequest) returns (AllocateResponse) {}
+}
+
+message RegisterRequest {
+	// Version of the API the Device Plugin was built against
+	string version = 1;
+	// Name of the unix socket the device plugin is listening on
+	string unixsocket = 2;
+}
+
+message RegisterResponse {
+	// Minimum version the Kubelet API supports.
+	string version = 1;
+	// Kubelet fills this field if it encounters any errors
+	// during the registration process or discover process
+	string error = 2;
+}
+
+message AllocateRequest {
+	repeated Device devices = 1;
```

Comment:
Mostly to stay consistent across the API. This also allow us to not have to update this if we allow for multiple Device Kind per Device Plugin.

It's also a lot cleaner because we aren't depending on the type of ID